app/components/search/SearchResultUsageNotes.tsx

```diff
@@ -0,0 +1,40 @@
+import { Link } from "@remix-run/react"
+import React from "react"
+import type JSXNode from "~/types/JSXNode"
+import SanitizedTextSpan from "../SanitizedTextSpan"
+
+interface SearchResultProps {
+  text: string
+  data: any[]
+}
+
+const SearchResultUsageNotes = ({ text, data }: SearchResultProps): JSXNode => {
+  if (data === undefined || data.length === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <h3 className="text-xl font-bold">
+        <>
+          Usage Notes containing &ldquo;{text}&rdquo;: {data.length}
+        </>
+      </h3>
+      {data.map((e) => {
+        return (
+          <div className="my-2 flex flex-col" key={e.id}>
+            <Link
+              to={`/entries/${e.headword}`}
+              className="font-bold text-red-600 hover:text-red-400"
+            >
+              {e.headword}
+            </Link>
+            <SanitizedTextSpan text={e.partofspeech + " \u2014 " + e.usage} />
+          </div>
+        )
+      })}
+    </>
+  )
+}
+
+export default SearchResultUsageNotes
```

Comment:
Re: all the `SearchResult______` components: they seem to all substantially be doing the same (or similar) things. Can we factor out the common behaviour, and possibly use a nested approach...? Something like this (which is kinda pseudo-JSX, haha):

```jsx
<SearchResults>
  {data.map((d) =>
    <SearchResult>
      <UsageNoteResult /> {/* Or <FistNoteResult />, etc. */}
    </SearchResult>
  }
</SearchResults>
```

Does this make sense?

Reply:
Yeah! That's a great way to do it

Reply:
OK for now, we will address later if there's more time.